### PR TITLE
PoSW diff adjustment

### PR DESCRIPTION
**Pull request by ping-ke**
Add fields to POSW 
Add function for PoSW diff adjustment

requirement: https://github.com/QuarkChain/QCEPs/pull/8

### CODE REVIEW

File: cluster/master/backend.go [hunk 1]
```diff
@@ -688,7 +688,7 @@ func (s *QKCMasterBackend) GetStats() (map[string]interface{}, error) {
 		shard["poswEnabled"] = powConfig.Enabled
 		shard["poswMinStake"] = powConfig.TotalStakePerBlock
 		shard["poswWindowSize"] = powConfig.WindowSize
-		shard["difficultyDivider"] = powConfig.DiffDivider
+		shard["difficultyDivider"] = powConfig.GetDiffDivider(shardState.Timestamp)
```
Comment (ninjaahhh) [inline cluster/master/backend.go:691]: we probably don't need boost for shard PoSW

File: core/minorblock_validator.go [hunk 2]
```diff
@@ -228,9 +228,9 @@ func (v *MinorBlockValidator) ValidateSeal(mHeader types.IHeader, usePowsDiff bo
 			return err
 		}
 	} else {
-		shardConfig := v.bc.shardConfig.PoswConfig
-		if shardConfig.Enabled {
-			adjustedDiff.Div(adjustedDiff, new(big.Int).SetUint64(shardConfig.DiffDivider))
+		config := v.bc.shardConfig.PoswConfig
+		if config.Enabled {
+			adjustedDiff.Div(adjustedDiff, new(big.Int).SetUint64(config.GetDiffDivider(header.GetTime())))
```
Comment (ninjaahhh) [inline core/minorblock_validator.go:233]: same as above

**Comment by qizhou** [comment]
Do we need to finalize the parameters?  Looks like we need 10000x following QCEP-8

File: cluster/config/config.go [hunk 3]
```diff
@@ -64,31 +64,65 @@ func NewPOWConfig() *POWConfig {
...
-	DiffDivider        uint64   `json:"DIFF_DIVIDER"`
-	WindowSize         uint64   `json:"WINDOW_SIZE"`
-	TotalStakePerBlock *big.Int `json:"TOTAL_STAKE_PER_BLOCK"`
+	Enabled               bool     `json:"ENABLED"`
+	EnableTimestamp       uint64   `json:"ENABLE_TIMESTAMP"`
+	DiffDivider           uint64   `json:"DIFF_DIVIDER"`
+	WindowSize            uint64   `json:"WINDOW_SIZE"`
+	TotalStakePerBlock    *big.Int `json:"TOTAL_STAKE_PER_BLOCK"`
+	BoostTimestamp        uint64   `json:"BOOST_TIMESTAMP"`
+	BoostMultiplerPerStep uint64   `json:"BOOST_MULTIPLER_PER_STEP"`
```
Comment (ninjaahhh) [inline cluster/config/config.go:73]: typo: boost multiplier

**Pull request merged**